contracts/LoanCore.sol

```diff
@@ -411,7 +408,7 @@ contract LoanCore is
         address lender = lenderNote.ownerOf(loanId);
 
         // if the loan has been completely repaid and no more repayments are expected
-        if (loans[loanId].balance == 0) {
+        if (loans[loanId].state == LoanLibrary.LoanState.Repaid) {
```

Comment:
Yes, this scenario would mean the lender couldn't receive those partial repayments.

For a fix, I think we could move away from having the holder of the lenderNote be able to claim and instead track for repayment claims with a mapping like this `noteReceipts[loanId][lender]`. This way even if a lender transfers the note, they can still collect payment from before they transfer it in the event it is burned/sold/transferred. What do you think?

With this, and the other prorated changes, we should update the token images to reflect the new LoanData params like interest paid, balance, last payment time, etc.. In order to provide a better experience when selling the Borrower/LenderNotes

Reply:
Definitely agree on the token images.

I think it's a matter of taste and user protection; I don't think we should move away from the LenderNote, that increases complexity. Right now we can say that the holder of the note holds all rights and obligations associated with the lending side, but if we move away, that would no longer be true, and we'd need to start putting "except..." in a lot of places.

I think it's a matter of taste here, should we protect users or not? If so, maybe we can just add a revert check in `claim` if there are unclaimed funds associated with the note? Or pay them out automatically on claim as well?

Reply:
I am a fan of adding the revert statement saying the lender should collect their payment before claiming the asset. This seems like the most straightforward approach. We could combine the `redeemNote` logic in the claim flow, but I think it would bloat the regular liquidation flow where payments are sent directly to the lender. Best to keep the happy path as efficient as possible imo

Reply:
Yep, agreed - think that sounds good to me!